UOP1_Project/Assets/Scripts/Pool/ComponentPool.cs

```diff
@@ -0,0 +1,55 @@
+﻿using UnityEngine;
+
+namespace OP1.Pool
+{
+	/// <summary>
+	/// Implements a Pool for Component types.
+	/// </summary>
+	/// <typeparam name="T">Specifies the component to pool.</typeparam>
+	public abstract class ComponentPool<T> : Pool<T> where T : Component, IPoolable
+	{
+		public abstract int InitialPoolSize { get; set; }
+		private GameObject _poolRootObject;
+
+		private void InitializePool()
+		{
+			_poolRootObject = new GameObject(name);
+			DontDestroyOnLoad(_poolRootObject);
+			for (int i = 0; i < InitialPoolSize; i++)
+			{
+				_available.Push(Add());
+			}
+		}
```

Comment:
I might be blind but, is this method called anywhere? Looks like `OnEnable` stuff but I don't see it 👀